### ISSUE TITLE
[a11y]: Dropdown - Layer structure is not conveyed 

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#16285]
### Package

@carbon/react

### Browser

Firefox

### Package version

v1.55.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

There are 3  Layers of  components  are present ,  while navigating through  these Dropdowns by  Tab , 
Screen reader users does not have any  indication about the Layering .

### Recommended fix

The Existing  Layer 1, Layer 2, & Layer 3  info must be associated with  connected Dropdowns  to indicate  the Layering  structure.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-dropdown--with-layer

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
